services/server.service.ts

```diff
@@ -329,6 +333,36 @@ export async function getAllProfilesForServer(
     .returns<ServerUserProfile>();
 }
 
-type GetAllProfilesForServerResponse = Awaited<ReturnType<typeof getAllProfilesForServer>>;
-export type GetAllProfilesForServerResponseSuccess = GetAllProfilesForServerResponse['data'];
-export type GetAllProfilesForServerResponseError = GetAllProfilesForServerResponse['error'];
+type GetAllProfilesForServerResponse = Awaited<
+  ReturnType<typeof getAllProfilesForServer>
+>;
+export type GetAllProfilesForServerResponseSuccess =
+  GetAllProfilesForServerResponse['data'];
+export type GetAllProfilesForServerResponseError =
+  GetAllProfilesForServerResponse['error'];
+
+export async function updateServerIcon(
+  supabase: SupabaseClient<Database>,
+  filePath: string,
+  image: any,
+  server_id: number
+) {
+  const { data, error } = await supabase.storage
+    .from('servericons')
+    .update(filePath, image, { upsert: true, cacheControl: '1' });
```

Comment:
Is there no way to update and return the new address here by the way? Feels wasteful to perform two operations like this